### PR TITLE
Molecule implementation

### PR DESCRIPTION
**Pull request by sabre1041**
#### What does this PR do?
Incorporates [Molecule](https://molecule.readthedocs.io) testing

#### How should this be tested?
Follow the steps outlined in the README in the molecule directory

#### Is there a relevant Issue open for this?
No

#### Who would you like to review this?
cc: @redhat-cop/openshift-applier @oybed

### CODE REVIEW

File: molecule/default/prepare.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        dest: "/usr/local/bin"
+        remote_src: yes
+    - name: Manage OpenShift Configuration File
+      block:
+        - name: Copy OpenShift Configuration File
+          copy:
+            src: "~/.kube/config"
+            dest: "~/.kube/"
+      when: use_kube_file|bool
+    - name: Manage OpenShift Configuration File via Login
```
Comment (oybed) [inline molecule/default/prepare.yml]: Any chance we can leverage the [openshift-login role](https://github.com/redhat-cop/casl-ansible/tree/master/roles/openshift-login)  ?

Comment (sabre1041) [reply molecule/default/prepare.yml]: With regard to the indenting, yes we are violating the indentation rule as the rules that are being used by yamllint extend the default set of rules. If we are to expand molecule and or linting across the set of repositories, additional configurations would need to be made to go against most yaml policies. I would recommend that we revisit the topic.

File: molecule/README.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+The following requirements must be met:
+
+* Proper [installation and configuration of Molecule](https://molecule.readthedocs.io/en/latest/installation.html)
+* Docker
+** Python docker dependencies
+
+## OpenShift Environment
+
+The primary goal of the `openshift-applier` role is to configure an OpenShift cluster. An existing environment must be available for the role to be executed against. The location and credentials are driven by environment variables. Authentication can be provided in the following forms:
```
Comment (etsauer) [inline molecule/README.md]: `s/configure an OpenShift cluster/manage resources in an OpenShift cluster/`

Comment (etsauer) [reply molecule/README.md]: `s/existing environment/existing OpenShift cluster/

**Pull request merged**